contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt

```diff
@@ -0,0 +1,124 @@
+        }
+        assertFailsWith(transaction,"This transaction should only output one BasketOfApples state")
+    }
+
+    @Test
+    fun blankDescription(){
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(
+                BasketOfApples(
+                    "",
+                    outputBasketOfApplesStateFarm,
+                    outputBasketOfApplesStateOwner,
+                    outputBasketOfApplesStateWeight,
+                    outputBasketOfApplesStateParticipants
+                )
+            )
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction,"The output BasketOfApples state should have clear description of Apple product")
+    }
+
+    @Test
+    fun basketWeightIsZero(){
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(
+                BasketOfApples(
+                    outputBasketOfApplesStateDescription,
+                    outputBasketOfApplesStateFarm,
+                    outputBasketOfApplesStateOwner,
+                    0,
+                    outputBasketOfApplesStateParticipants
+                )
+            )
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction,"The output BasketOfApples state should have non zero weight")
+    }
+
+    @Test
+    fun basketWeightIsNegative(){
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(
+                BasketOfApples(
+                    outputBasketOfApplesStateDescription,
+                    outputBasketOfApplesStateFarm,
+                    outputBasketOfApplesStateOwner,
+                    -1,
+                    outputBasketOfApplesStateParticipants
+                )
+            )
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction,"The output BasketOfApples state should have non zero weight")
+    }
+
+    @Test
+    fun missingCommand(){
+        val transaction = buildTransaction {
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction, "List is empty.")
+    }
+
+    @Test
+    fun unknownCommand(){
+        class MyDummyCommand: Command
+        val transaction = buildTransaction {
+            addCommand(MyDummyCommand())
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction,
+            "Incorrect type of BasketOfApples commands: com.r3.developers.apples.contracts." +
+                    "BasketOfApplesContractPackBasketCommandTest\$unknownCommand\$MyDummyCommand"
```

Comment:
Maybe the contract testing lib needs to have some kind of `assertFailsWithMessageContaining` to allow you to only specify part of the message.

Reply:
Oooh that'd a good feature!! 😁 I'll have a think about what people would want in the background

Reply:
https://r3-cev.atlassian.net/browse/CORE-15193